builder/openstack/step_source_image_info.go

```diff
@@ -0,0 +1,96 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/gophercloud/gophercloud/pagination"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type StepSourceImageInfo struct {
+	SourceImage     string
+	SourceImageName string
+	ImageFilters    ImageFilterOptions
+}
+
+type ImageFilterOptions struct {
+	Filters    map[string]interface{} `mapstructure:"filters"`
+	MostRecent bool                   `mapstructure:"most_recent"`
+}
+
+func (s *StepSourceImageInfo) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(Config)
+	ui := state.Get("ui").(packer.Ui)
+
+	client, err := config.computeV2Client()
+
+	// if an ID is provided we skip the filter since that will return a single or no image
+	if s.SourceImage != "" {
+		state.Put("source_image", s.SourceImage)
+		return multistep.ActionContinue
+	}
+
+	params := &images.ListOpts{}
+
+	// build ListOpts from filters
+	if len(s.ImageFilters.Filters) > 0 {
+		err = buildImageFilters(s.ImageFilters.Filters, params)
```

Comment:
This part does not seems to work for me because this function never returns `nil` what I can see. The function returns a packer.MultiError so I changed it to the below:

```
multiErr := buildImageFilters(s.ImageFilters.Filters, params)
if len(multiErr.Errors) > 0 {
  err := fmt.Errorf("Errors encountered in filter parsing.\n%s" + multiErr.Error())
```
Do not know it this is the way to handle it but then I get passed this piece of code.

Reply:
Yeah I'm handling it wrong, as an Error but it's an MultiError. Handling it your way would work due to the nil slice length being 0.